speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/DatabaseSettingsMenu.cpp

```diff
@@ -0,0 +1,261 @@
+#include <tgfclient.h>
+#include "guimenu.h"
+#include "legacymenu.h"
+#include "Mediator.h"
+#include "DatabaseSettingsMenu.h"
+#include "DataSelectionMenu.h"
+// Parameters used in the xml files
+#define PRM_USERNAME "UsernameEdit"
+#define PRM_PASSWORD "PasswordEdit"
+#define PRM_ADDRESS  "AddressEdit"
+#define PRM_PORT     "PortEdit"
+#define PRM_SCHEMA   "SchemaEdit"
+#define PRM_SSL      "CheckboxUseSSL"
+#define PRM_CERT     "dynamic controls/CertificateSettings"
+#define PRM_DBSTATUS "DbStatus"
+
+// GUI screen handles
+static void* s_scrHandle = nullptr;
+static void* s_nextHandle = nullptr;
+
+// Control variables for the menu
+int m_usernameControl;
+int m_passwordControl;
+int m_addressControl;
+int m_portControl;
+int m_schemaControl;
+int m_useSSLControl;
+int m_dbStatusControl;
+
+char m_portString[256];
+
+tDatabaseSettings m_dbsettings;
+
+/// @brief Handle input in the Username textbox
+static void SetUsername(void*)
+{
+    sprintf(m_dbsettings.Username, GfuiEditboxGetString(s_scrHandle, m_usernameControl));
+    GfuiEditboxSetString(s_scrHandle, m_usernameControl, m_dbsettings.Username);
+}
+
+/// @brief Handle input in the Password textbox
+static void SetPassword(void*)
+{
+    sprintf(m_dbsettings.Password, GfuiEditboxGetString(s_scrHandle, m_passwordControl));
+    GfuiEditboxSetString(s_scrHandle, m_passwordControl, m_dbsettings.Password);
+}
+
+/// @brief Handle input in the Address textbox
+static void SetAddress(void*)
+{
+    sprintf(m_dbsettings.Address, GfuiEditboxGetString(s_scrHandle, m_addressControl));
+    GfuiEditboxSetString(s_scrHandle, m_addressControl, m_dbsettings.Address);
+}
+
+/// @brief Handle input in the Port textbox
+static void SetPort(void*)
```

Comment:
Make the port edit-box only accept numbers just like the edit-boxes in the controls menu